langchain/prompt.py

```diff
@@ -84,3 +84,102 @@ def from_examples(
         example_str = example_separator.join(examples)
         template = prefix + example_str + suffix
         return cls(input_variables=input_variables, template=template)
+
+
+class DynamicPrompt(BaseModel):
```

Comment:
probably want to a BasePrompt class that just exposes the interface (`format`) so we can just use that for type hints everywhere